internal/keys/hasher.go

```diff
@@ -58,6 +59,16 @@ func (t tupleKeysHasher) Append(h hasher) error {
 			return sortedTupleKeys[i].GetUser() < sortedTupleKeys[j].GetUser()
 		}
 
+		cond1 := sortedTupleKeys[i].GetCondition()
+		cond2 := sortedTupleKeys[j].GetCondition()
+		if cond1 == nil && cond2 == nil {
+			return true
+		} else if cond1.GetName() != cond2.GetName() {
```

Comment:
```suggestion
		if cond1 != nil && cond2 != nil && cond1.GetName() != cond2.GetName() {
```

Reply:
This is a slight change from the current behavior when only one condition is nil, if we want to retain sort order there we can tweak things slightly:
```suggestion
if (cond1 != nil || cond2 != nil) && cond1.GetName() != cond2.GetName() {
```